packages/components/src/DataTable/Header/DataTableHeaderCell.tsx

```diff
@@ -78,9 +80,10 @@ const DataTableHeaderCellLayout = forwardRef(
             : 'none'
         }
         className={className}
-        onClick={handleClick}
         ref={ref}
         style={{ cursor: canSort ? 'pointer' : undefined }}
+        tabIndex={-1}
+        {...clickableProps}
```

Comment:
I forgot that `clickableProps` contains `tabIndex: 0`. This change should fix the tab moving focus around the header issue:
```suggestion
        {...clickableProps}
        tabIndex={-1}
```